azext_iot/deviceupdate/commands_instance.py

```diff
@@ -52,18 +52,29 @@ def create_instance(
 
 def update_instance(cmd, parameters: DeviceUpdateMgmtModels.Instance):
     from azext_iot.deviceupdate.common import ADUInstanceDiagnosticStorageAuthType
+
     instance_manager = DeviceUpdateInstanceManager(cmd=cmd)
     storage_properties = parameters.diagnostic_storage_properties
+    if storage_properties:
+        is_dict = isinstance(storage_properties, dict)
```

Comment:
This logic is not because of the service response, it is how the CLI generic update helpers assemble the properties of the this Track 2 SDK model. I expect this to change significantly when integrating with model-less client. If a new object is being created the model attribute (storage_properties) becomes a dict, if a non-object attribute of an existing model is modified it is handled in-place.